packages/match-media/src/index.js

```diff
@@ -4,19 +4,32 @@ import { useThemeUI } from 'theme-ui'
 // Shared with @styled-system/css
 const defaultBreakpoints = [40, 52, 64].map(n => n + 'em')
 
-export const useBreakpointIndex = () => {
+export const useBreakpointIndex = defaultIndex => {
   const context = useThemeUI()
   const breakpoints =
     (context.theme && context.theme.breakpoints) || defaultBreakpoints
 
-  const getIndex = useCallback(
-    () =>
-      breakpoints.filter(
-        breakpoint =>
-          window.matchMedia(`screen and (min-width: ${breakpoint})`).matches
-      ).length,
-    [breakpoints]
-  )
+  const getIndex = useCallback(() => {
+    if (typeof window === 'undefined') {
+      if (typeof defaultIndex === 'number') {
+        if (defaultIndex < 0 || defaultIndex > breakpoints.length - 1) {
+          throw new RangeError(
+            `Default breakpoint index out of range. Theme has ${breakpoints.length} breakpoints, got index ${defaultIndex}`
+          )
+        }
+        return defaultIndex
+      }
+      throw new TypeError(
+        'To use @theme-ui/match-media hooks on the server, you must pass a default index. Got: ' +
```

Comment:
Thanks for looking into this. I might've missed this part of the conversation, but why not default to `0` on the server? Seems like a sensible default IMO

Reply:
That was my initial thought until @dburles mentioned that it could cause unexpected re-rendering when loaded at larger screen sizes. Could be jarring for desktop users to see content replaced shortly after load

Reply:
My guess is that that would be an issue regardless. I think having the ability to change the default makes sense, but also that this should have a sensible default

Reply:
Oh - I misunderstood. A default for the default 😅 updated